### ISSUE TITLE
More verbose indexing and deleting when iterating data 

### ISSUE THREAD

**Issue by cristianmtr** [jina-ai/jina#2220]
### User story: 
As a user of Jina who is adding data to be Indexed via iteration, or deleting data via iteration. I would expect to be able to see the status of each document, which will allow me to correctly debug any errors that occur. This applies when index or deleting via the gPRC gateway (Python client). 

Currently, the following command only returns a true or false if all documents are indexed or not. 
```
with f:
    f.index((Document(f'doc content {i}') for i in range(10)))
```

A more user-friendly output would be 
```
with f:
    f.index((Document(f'doc content {i}') for i in range(10)))

Doc 1 - Indexed 
Doc 2 - Indexed 
Doc 3 - Failed, error 
```

### Acceptance Criteria:
This ticket is done when it is possible for a user to see which document in an iteration has been indexed or deleted correctly, and which has failed. It would be ideal if the failure error was also returned to the user. 

### Old content
Part of adding ACID-compliance 

Tell the user what documents were successfully processed (insert, delete, update), which failed.

Workload for **basic** approach:
- all Indexers (Np, BinaryPb, Hub etc.) should keep track of what failed, what succeeded 
- all CRUD-related Drivers (VectorIndex, KVIndex etc.) should log

Reference back to https://github.com/jina-ai/jina/issues/2165

A future task will be  Drivers also perform cleanup tasks

example: if partial Index success, delete the documents that were inserted.

**Comment by florian-hoenicke** [comment]
I think we should also make sure that the error is traversed back to the client. The most common use case will be the REST interface used by the console or some flow manager. If we just log the errors, the user won't see them.
It can be a follow-up implementation.

**Comment by FionnD** [comment]
Good point @florian-hoenicke , I would see that a AC of this ticket to be honset

**Comment by FionnD** [comment]
Question: 
If a user sends a batch of files to be indexed via the PostgreSQL indexer. How the indexer handles failures in this batch, and are they correctly informed to the user? For example, in this situation, you send a batch of ten files. The seventh file fails to index. Have the first six been correctly indexed and do you need to re-try to index the last three? or have they all failed? 


Once we understand the logic, then we need to understand how this is currently logged to the user. 

`File 1 - Indexed successfully 
File 2 - Failed to index. Please re-try`

**Comment by cristianmtr** [comment]
Yes, good point. One way to test this would be to add 2 documents with the same id within the same batch. Since PSQL has a primary key uniqueness constraint, it will fail. It would be worth checking to see how it fails. Does it save the others?

We also do need provide better logging

**Comment by FionnD** [comment]
@JoanFM @maximilianwerk Following our chat in the meeting this morning, can both of you please look at the user story of this ticket and determine if we need to now work on this now or after the API refactoring?

**Comment by JoanFM** [comment]
I would say after API refactoring. what do u think @maximilianwerk

**Comment by FionnD** [comment]
@cristianmtr

**Comment by cristianmtr** [comment]
Partly moved to be covered in https://github.com/jina-ai/indexers/issues/14
